pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -134,7 +141,17 @@ type amiFilter struct {
 	owner       string
 }
 
-func getDefaultAMIID(client *ec2.EC2, os providerconfig.OperatingSystem) (string, error) {
+func getDefaultAMIID(client *ec2.EC2, os providerconfig.OperatingSystem, region string) (string, error) {
+	cacheLock.Lock()
+	defer cacheLock.Unlock()
+
+	cacheKey := fmt.Sprintf("ami-id-%s-%s", region, os)
+	amiID, found := cache.Get(cacheKey)
+	if found {
+		glog.V(4).Info("found AMI-ID in cache!")
+		return amiID.(string), nil
+	}
+
 	filter, osSupported := amiFilters[os]
```

Comment:
I think the check if the os is supported should be the first one

Reply:
Done with https://github.com/kubermatic/machine-controller/pull/391/commits/34fa383d134719058f7b065f840f75bcf4b9b43a